rust/arrow/src/datatypes.rs

```diff
@@ -1836,7 +1836,7 @@ impl Schema {
     /// ```
     /// use arrow::datatypes::*;
     ///
-    /// let merged = Schema::try_merge(&vec![
+    /// let merged = Schema::try_merge(vec![
```

Comment:
this example shows a pretty good example of how I think the usability (as well as performance) of this API is improved by this PR